core/src/main/java/nu/yona/server/messaging/service/MessageService.java

```diff
@@ -189,14 +197,24 @@ private void deleteMessages(Collection<Message> messages)
 		involvedMessageDestinations.forEach(d -> MessageDestination.getRepository().save(d));
 	}
 
-	private MessageSource getNamedMessageSource(UserDto user)
+	private MessageSource getNamedMessageSource(User user)
 	{
-		return MessageSource.getRepository().findOne(user.getOwnPrivateData().getNamedMessageSourceId());
+		return getMessageSource(user.getNamedMessageSourceId());
 	}
 
 	private MessageSource getAnonymousMessageSource(UserDto user)
 	{
-		return MessageSource.getRepository().findOne(user.getOwnPrivateData().getAnonymousMessageSourceId());
+		return getMessageSource(user.getOwnPrivateData().getAnonymousMessageSourceId());
+	}
+
+	private MessageSource getAnonymousMessageSource(User user)
+	{
+		return getMessageSource(user.getAnonymousMessageSourceId());
+	}
+
+	private MessageSource getMessageSource(UUID id)
+	{
+		return MessageSource.getRepository().findOne(id);
```

Comment:
It is better to add the `MessageSourceRepository` as a dependency.

Reply:
Agreed.